src/functions/guild/modifyRolePerms.js

```diff
@@ -7,35 +7,35 @@ module.exports = async d => {
     const [guildID, roleID, ...perms] = data.inside.splits;
 
     const guild = await d.util.getGuild(d, guildID);
-    if (!guild) return d.aoiError.fnError(d, 'guild', {inside: data.inside});
+    if (!guild) return d.aoiError.fnError(d, 'guild', { inside: data.inside });
 
     const role = await guild.roles.fetch(roleID).catch(e => undefined);
-    if (!role) return d.aoiError.fnError(d, 'role', {inside: data.inside});
+    if (!role) return d.aoiError.fnError(d, 'role', { inside: data.inside });
+
+    let newPerms = BigInt(0);
 
-    let arrayPerms = role.permissions.toArray(); // make array array
     if (perms.includes('+all')) {
-        arrayPerms = ['Administrator']; 
+        newPerms = Permissions.ALL;
     } else if (perms.includes('-all')) {
-        arrayPerms = [];
+        newPerms = BigInt(0);
     } else {
         for (const perm of perms) {
             const sign = perm.slice(0, 1);
+            const permission = Permissions[perm.slice(1)];
+
             if (sign === '+') {
-                arrayPerms.push(Permissions[perm.slice(1)]); 
+                newPerms |= BigInt(permission); // big int go brr
             } else if (sign === '-') {
-                const index = arrayPerms.indexOf(Permissions[perm.slice(1)]);
-                if (index !== -1) {
-                    arrayPerms.splice(index, 1); 
-                }
+                newPerms &= ~BigInt(permission); // same abov
```

Comment:
This causes the role to lose all previous permissions, and only setting the new perms.